pkg/credentials/provider.go

```diff
@@ -34,7 +37,7 @@ func NewProvider(audience, roleARN string, jwtSource JWTSource) (*Provider, erro
 type Provider struct {
 	credentials.Expiry
 
-	stsClient *sts.STS
+	stsClient stsiface.STSAPI
 	jwtSource JWTSource
```

Comment:
We've migrated to using the STS interface (STSAPI) so that we can support mocking.